src/moduleResolver/asyncModuleResolver.ts

```diff
@@ -66,8 +66,11 @@ export async function asyncModuleResolver(ctx: Context, entryFiles: Array<string
       }
       let pkg = bundleContext.getPackage(resolved.package.meta);
 
+      const isUserPackage = resolved.package.isUser;
+      const pkgType = isUserPackage ? PackageType.USER_PACKAGE : PackageType.EXTERNAL_PACKAGE;
+
       if (!pkg) {
-        pkg = createPackage({ meta: resolved.package.meta, type: PackageType.EXTERNAL_PACKAGE });
+        pkg = createPackage({ meta: resolved.package.meta, type: pkgType });
```

Comment:
This is the main change for the user/non-user detection.  We set the type to `USER_PACKAGE` based on whether `isUser` was set during package resolution.

Here I am understanding a package to be a "User" package if it is expected that the user will modify the package.  So this should be true for custom modules workspace packages, but false for registry-downloaded packages.